packages/gitbook/src/components/DocumentView/CodeBlock/CodeBlockRenderer.tsx

```diff
@@ -0,0 +1,272 @@
+import { DocumentBlockCode, JSONDocument } from '@gitbook/api';
+
+import { tcls } from '@/lib/tailwind';
+
+import { CopyCodeButton } from './CopyCodeButton';
+import type { HighlightLine, HighlightToken } from './highlight';
+import { AnnotationPopover } from '../Annotation/AnnotationPopover';
+import { BlockProps } from '../Block';
+
+import './theme.css';
+
+type CodeBlockRendererProps = Pick<
+    BlockProps<DocumentBlockCode>,
+    'block' | 'document' | 'style'
+> & {
+    lines: HighlightLine[];
+};
```

Comment:
It's almost either to just write:

```suggestion
type CodeBlockRendererProps = {
    block: DocumentBlockCode;
    document: DocumentJSON;
    style: StyleValue;
    lines: HighlightLine[];
};
```

Reply:
💥  by doing this, you can notice that you are passing `document` which means: every client code blocks gets the entire document JSON serialized by Next to be passed as a prop.

We can't do that, it'll multiply by x10/x100 the HTML size.

Reply:
When you use that notation, JSDOC comments are not preserved, it's always better to use `Pick`.